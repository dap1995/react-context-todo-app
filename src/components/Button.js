import Link from './Link';

const Button = Link.withComponent('button').extend`
  &:focus {
    outline: 0;
    border:1px solid rgba(175,47,47,0.15);
    box-shadow: 0 0 10px #e6e6e6;
  }
`;

export default Button;
