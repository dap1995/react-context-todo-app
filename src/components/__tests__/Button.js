import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button onClick={() => {}} />, div);
  });

  it('should be the same as the snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should hide button when hidden prop is true', () => {
    const tree = renderer.create(<Button hidden={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
