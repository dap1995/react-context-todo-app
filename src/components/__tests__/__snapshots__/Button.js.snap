// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be the same as the snapshot 1`] = `
<button
  className="sc-htpNat fjeIch"
/>
`;
